providers/faux/faux.go

```diff
@@ -43,9 +44,10 @@ func (p *Provider) BeginAuth(state string) (goth.Session, error) {
 func (p *Provider) FetchUser(session goth.Session) (goth.User, error) {
 	sess := session.(*Session)
 	return goth.User{
-		UserID:          sess.ID,
-		Name:            sess.Name,
-		Email:           sess.Email,
+		UserID:   sess.ID,
+		Name:     sess.Name,
+		Email:    sess.Email,
+		Provider: sess.Provider,
```

Comment:
Not sure what you mean here?
There is no need to store the Provider inside of the session, since retrieving the session is already based on the provider name and so you will always know that.

So think it is better to remove it